crowbar_framework/app/models/service_object.rb

```diff
@@ -1305,6 +1305,19 @@ def apply_role(role, inst, in_queue, bootstrap = false)
         node_attr_cache[node_name]["windows"]
       end
 
+      # refill pre_cached_nodes in one call instead of allowing
+      # #remote_chef_client_threads to load node by node, thus speeding it up
+      search = ""
+      nodes_to_run.each do |n|
+        search << "name:#{n}" if n.equal? nodes_to_run.first
```

Comment:
why not do something like `search = nodes_to_run.join(" OR ")` ?

Reply:
we would still need to do something extra for the first as the format is "name:NODE1 OR name:NODE2 OR name:NODE3"

Reply:
like this `search - (nodes_to_run.map { |n| "name:"+n }).join(" OR ")` ?

Reply:
nice! changing it!